### PR TITLE
Reorganizing

### PR DESCRIPTION
**Pull request by ElPhysico**
Additional to the discussed changes, I further implemented the following points:

- Input changes have been reverted, input or -i option will now screen for input file type. The input files in "molecule" folder have been adjusted to be loaded correctly.
- Variable names have been changed (no more Anne, I'll miss her).
- Original output format can now be changed to csv format by using -csv option.
- Console output can now be controlled globally. Minimal console output remains nevertheless. Warnings and Errors will still be written to console.
- Minor quality changes.

Additionally, the default.config should be adjusted to create the initial results that Anne's code was producing.

### CODE REVIEW

File: src/funcs.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include "funcs.hpp"
+
+
+/* Functions */
+void progresssBar(double progress, std::string msg)
```
Comment (takenbymood) [inline src/funcs.cpp:5]: This should be turned off for now to avoid compatibility issues

File: src/main.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#include <algorithm>        /* std::remove */
+#include <iomanip>          /* std::setw() */
+#include <sys/stat.h>       /* chmod */
+
+
+typedef std::chrono::time_point<std::chrono::high_resolution_clock> time_point;
+
+
+/* Classes and structures */
+struct filePaths_ {
```
Comment (takenbymood) [inline src/main.hpp:20]: Can we make these member variables static? Initialisation of non-static class or struct variables is a C++11 feature. If that's not possible then it's fine but we should add the C++11 requirement to the makefile

File: src/xyz.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  //   fprintf(outf, "\t%.8f", fib.mol.diameterAtom);
+  // }
+  // //
+  // for (int i = 0; i < fib.mol.numAtoms; i++) {
+  //   fprintf(outf, "\n4");
+  //   fprintf(outf, "\t%.8f", (-box + 0 * fib.offset + i * fib.mol.distanceAtoms) / res);
+  //   fprintf(outf, "\t%.8f", 0.);
+  //   fprintf(outf, "\t%.8f", -fib.mol.diameterAtom);
+  // }
+
```
Comment (takenbymood) [inline src/xyz.cpp]: Are all the commented lines left in for future? Also there's a lot of repeated printing things like "\n4" and "%t.8f",0. could this be handled in a "printtofile" function to save on repeated lines? Not urgent.

File: src/entrepot.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
+// // fib.radGap = 34.75;
```
Comment (takenbymood) [inline src/entrepot.cpp]: Is it worth keeping this file in right now? If it's good to have then we can keep it, but would be nice to keep commented code to a minimum I think? Obviously always going to be some here and there, I do it all the time 😓

**Review by takenbymood — CHANGES_REQUESTED** [review]
Looks good, just a few comments on the files below :)

Only major concern: output was slightly different running this version of the code compared to the previous one. Is there some randomness involved that could account for this or do we need to track this down? Is it possible that the header information is not correct?

#atoms per molecule N = 36
#distance_atoms = 0.255
#molecule_length = 8.925
#layers = 2

#LJ_epsilon	CD_epsilon	lateral_gap_min	radial_gap_min	offset_min	D-periodicity_min	E_CD_min	E_LJ_min	E_total_min
0.100	20.000	0.784	4.845	6.885	6.885	-1321.668	205.391	-1116.277

#atoms per molecule N = 36
#distance_atoms = 0.255
#molecule_length = 8.925
#layers = 2

#LJ_epsilon	CD_epsilon	lateral_gap_min	radial_gap_min	offset_min	D-periodicity_min	E_CD_min	E_LJ_min	E_total_min
0.100	20.000	0.784	4.900	6.900	6.925	-1318.741	202.914	-1115.827

**Comment by ElPhysico** [comment]
**tl;dr at the bottom ;-)**

Regarding your main concern:

I just checked, the header information should update correctly.

I also ran the code with all default parameters and got exactly the same result as the ones in Anne's historic file that I always take as benchmark.

- Anne's:

#N_atomspermolecule = 36
#distance_atoms = 0.255
#molecule_length = 8.925
#LJ_epsilon 	 CD_epsilon 	 lateral_gap_min 	 radial_gap_min 	 offset_min 	 D-periodicity_min 	 E_CD_min 	 E_LJ_min 	 E_total_min
0.100		20.000			0.896			4.500		6.700		6.725			-261.426	17.053	-244.373

- Run with newest commit:

#atoms per molecule N = 36
#distance_atoms = 0.255
#molecule_length = 8.925
#layers = 2

#LJ_epsilon	CD_epsilon	lateral_gap_min	radial_gap_min	offset_min	D-periodicity_min	E_CD_min	E_LJ_min	E_total_min
0.100	20.000	0.896	4.500	6.700	6.725	-261.426	17.053	-244.373



There are **definitely** differences in the calculation, mainly caused by the lines 668-677 in the version on the main branch:
```
for (int lat = 0; lat <= lat_max; lat++) {
    lat_gap = lat * 0.1 * diameter_atom;
  for (int rad = 0; rad <= rad_max; rad++) {

    rad_gap = rad * distance_atoms;
    double tmp = (L - (layers - 1.0) * rad_gap) / layers / distance_atoms;
    off_max = (int) ceil(tmp);

    for (int off = 0; off <= off_max; off++) {
      offset = rad_gap + off * distance_atoms;
```
comparing this to the current version, which is identical to Anne's original calculations:
```
for (int lat = 0; lat <= latMax; lat++) {
    latGap_ = lat * 0.1 * mol.diameterAtom;
    for (int rad = 0; rad <= radMax; rad++) {
      radGap_ = rad * 0.1;
      offMax = (int) ceil((mol.length + radGap_) / 0.1);
      for (int off = 0; off <= offMax; off++) {
        offset_ = off * 0.1;
```
I find two minor differences and a major difference.
The minor ones are, that in the main branch version, _rad_gap_ and _offset_ are incremented by _distance_atoms_, whereas in the original version they are incremented by 0.1.
The major difference is the way how _off_max_ and _offset_ are actually calculated. I remember that I tried to speed up the process back then by avoiding impossible configurations when using more than 2 layers, that is why the _layers_ term is in the calculation. Now this might, or might not, throw the values off, but since your values are still comparably close, I guess my calculation wasn't entirely wrong.
But since the increments are already different in both versions, one should expect that the energy is measured at different values and hence, the minimum configuration will not be exactly at the same points. I hope this makes sense.

Now, what worries me more, is the fact that you get such huge energy terms. I tried different things to replicate them, but I can't come up with them. Even when using 5 layers the energy is not 4 digits.


**tl;dr: Slight differences in the min values of the parameters are to be expected because of different increment sizes used in both version of the codes, so nothing to worry about. However, the large energy terms that you get are concerning.**

**Since the energies are high in both version, did you check that you are using the correct input file, i.e. the D-mimetic configuration?**

File: src/funcs.cpp (hunk 1 above)
Comment (ElPhysico) [reply src/funcs.cpp:5]:
```suggestion
// void progresssBar(double progress, std::string msg)
```
This change is live in the newest commit.

File: src/main.hpp (hunk 2 above)
Comment (ElPhysico) [reply src/main.hpp:20]: Bear with me, I'm really just a pragmatic "programmer", so I'm not entirely sure what implications that would have. As far as I understand it, static variables will remember changes. So if I have two instances of _filePaths__ struct, and I change a static variable in the one instance, it will also be changed in the other instance?
I guess all variables in all structs would need to be static then?
That wouldn't be a problem until I want to use multiple instances of the molecule or fibril struct, which is actually the main reason I implemented structs, so I can use multiple instances.

On the other hand, you say that adding C++11 requirement to the makefile would be okay? Can we then just do that and maybe you can explain to me at some point if I understood static variables correctly ^^

File: src/xyz.cpp (hunk 3 above)
Comment (ElPhysico) [reply src/xyz.cpp]: I'm not even sure this file is still needed. Will probably delete it in on one of the next commits.

Comment (ElPhysico) [reply src/xyz.cpp]: This was a quick  thing to create snapshots for Andela that she requested last minute a few weeks ago for her talk

File: src/entrepot.cpp (hunk 4 above)
Comment (ElPhysico) [reply src/entrepot.cpp]: Actually this file should not be in there, it was just a place for me to copy stuff to that I might want to reuse in the future. I'll save it to a folder that is ignored by git.

File: src/xyz.cpp (hunk 3 above)
Comment (ElPhysico) [reply src/xyz.cpp]: Removed it. Live in the newest commit.

File: src/entrepot.cpp (hunk 4 above)
Comment (ElPhysico) [reply src/entrepot.cpp]: Removed it. Live in the newest commit.

File: src/main.hpp (hunk 2 above)
Comment (ElPhysico) [reply src/main.hpp:20]: I added c++11 requirement to Makefile, should this be a problem, let me know.
